app/cdap/components/DirectiveInput/index.tsx

```diff
@@ -0,0 +1,151 @@
+import styled from 'styled-components';
+
+const SearchBarWrapper = styled(Box)`
+  padding: 10px;
+  display: flex;
+  justify-content: space-between;
+`;
+
+const InputWrapper = styled(Box)`
+  width: 100%;
+`;
+
+const UsageDirectiveWrapper = styled(Box)`
+  background: #616161;
+  box-shadow: -3px -4px 15px rgba(68, 132, 245, 0.25);
+`;
+
+export default function({
+  columnNamesList,
+  onDirectiveInputHandler,
+  onClose,
+  openDirectivePanel,
+}: IDirectiveInputProps) {
+  const [directiveInput, setDirectiveInput] = useState<string>('');
+  const [isColumnSelected, setIsColumnSelected] = useState<boolean>(false);
+  const [isDirectiveSelected, setIsDirectiveSelected] = useState<boolean>(false);
+  const [usageDirective, setUsageDirective] = useState<IUsageDirectives[]>([]);
+  const [directivesList, setDirectivesList] = useState<IDirectivesList[]>([]);
+  const directiveRef = useRef();
+  const classes = useStyles();
+
+  const handleDirectiveChange = (event: IOnRowClickValue | React.ChangeEvent<HTMLInputElement>) => {
+    if (!event.target.value) {
+      setIsDirectiveSelected(false);
+    }
+    setDirectiveInput(event.target.value);
+  };
+
+  useEffect(() => {
+    setUsageDirective(formatUsageDirectiveData(directiveInput, directivesList));
+  }, [directiveInput]);
+
+  const onEnterClick = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    {
+      const usageArraySplit =
+        usageDirective.length > 0 ? usageDirective[0]?.item?.usage.split(' ') : [];
+      const inputSplit = directiveInput.replace(/^\s+/g, '').split(' ');
+      if (event.key === 'Enter' && handlePasteDirective(directiveInput, directivesList)) {
+        onDirectiveInputHandler(directiveInput);
+      } else if (
+        event.key === 'Enter' &&
+        isColumnSelected &&
+        isDirectiveSelected &&
+        (usageArraySplit.length === inputSplit.length || inputSplit.length > usageArraySplit.length)
+      ) {
+        onDirectiveInputHandler(directiveInput);
+      }
+    }
+  };
+
+  return (
+    <>
+      {openDirectivePanel && (
+        <Box data-testid="directive-input-parent">
+          <AutoCompleteList
+            directiveInput={directiveInput}
+            onRowClick={(eventObject) => handleDirectiveChange(eventObject)}
+            setDirectivesList={setDirectivesList}
+            getDirectiveUsage={(activeResults: IUsageDirectives[], value) => {
+              setIsDirectiveSelected(value);
+              setUsageDirective(activeResults);
+            }}
+            onColumnSelected={() => {
+              setIsColumnSelected(true);
+            }}
+            isDirectiveSelected={isDirectiveSelected}
+            columnNamesList={columnNamesList}
+          />
+          <UsageDirectiveWrapper>
+            {Array.isArray(usageDirective) && usageDirective.length === 1 ? (
+              usageDirective.map((row: IUsageDirectives) => {
+                return <UsageDirective row={row} />;
```

Comment:
why return statemet, directly ()=> () as it is arriw function


also, add unique key to each element in array loop, map

Reply:
Done